### PR TITLE
docs(core,platform): add important notest related to i18n, formats, validation to date time pickers components documentation

### LINKED ISSUE #4880: Platform date picker inconsistent in the use of locale for input validation
#### Is this a bug, enhancement, or feature request?
Bug in fundamental-ngx/platform date picker.

#### Briefly describe your proposal.
Date Picker input seems to default to American-style (month/day/year) format to validate dates typed in the date picker, even though it uses the specified locale to display the date picked with the picker.

The locale should be used for both input and display formats. 

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
As per the stackblitz in the documentation (Angular 10.2.3, Fundamental 0.28).

#### If this is a bug, please provide steps for reproducing it.
Go to 
https://sap.github.io/fundamental-ngx/#/platform/date-picker
and scroll down to 'Internationalization of Date Picker'.

Click on the calendar and pick 'January 13 2020'. Calendar displays:
13/01/2020 (correct).

Then type 14/01/2020.
Input field will be highlighted in red. I understand because the input format is month/day/year and month 14 would be invalid.

#### Please provide relevant source code if applicable.
https://sap.github.io/fundamental-ngx/#/platform/date-picker
and scroll down to 'Internationalization of Date Picker'.

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue. 
closes #4880 

## Description

Because of limited support of i18n features based on native javascript functionality we have problems with these features.
Moment date-time adapter was implemented to solve these issues.
This PR contains changes related to documentation, shows important information related to the new feature, and explains the problems with the default fundamental adapter.

Reopened this [PR](https://github.com/SAP/fundamental-ngx/pull/6351)

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [n/a] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [n/a] check for console logs (warnings, errors)
-   [n/a] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/time-picker/time-picker-header/time-picker-header.component.html [hunk 1]
```diff
@@ -1,17 +1,41 @@
...
+        <br />
+        Since this adapter has some disadvantages, formatting may vary among browsers and parsing is limited to en
+        locale only.
+        <br />
+        You can use a custom fundamental adapter based on a third party library:
+        <a href="https://momentjs.com/" target="_blank">https://momentjs.com/</a>.
+        <br />
+        Please check examples and documentation on:
+        <b> <a routerLink="/core/moment-datetime-adapter" target="_blank"> Moment date time adapter </a> </b>.
+    </fd-message-strip>
```
Comment (dimamarksman) [inline apps/docs/src/app/core/component-docs/time-picker/time-picker-header/time-picker-header.component.html:38]: can we keep it in a sharable component so we don't need to repeat it on every date/time-related pages?

**Review by dmitry-stepanenko — APPROVED** [review]

**Review by dimamarksman — APPROVED** [review]

**Comment by puru-hk** [comment]
Looks fine

**Review by puru-hk — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/datetime-picker/datetime-picker-docs.component.html [hunk 2]
```diff
@@ -84,6 +84,28 @@
 
 <fd-docs-section-title id="complex-i18n" componentName="datetime-picker"> i18n example. </fd-docs-section-title>
 <description>
+    <fd-message-strip [dismissible]="false" type="information">
```
Comment (mikerodonnell89) [inline apps/docs/src/app/core/component-docs/datetime-picker/datetime-picker-docs.component.html:87]: why not use the `fd-datetime-important` component here?

**Review by mikerodonnell89 — APPROVED** [review]
Looks good, but let's use `fd-datetime-important` in core's datetime-picker-docs as well

**Pull request merged**